regex.py

```diff
@@ -0,0 +1,14 @@
+
+
+import re
+
+test_str = 'hi there 255,255,255,255'
+
+pattern = re.compile(r"(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?),(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?),(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?),(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)")
```

Comment:
Nice, I like the regex! The suggestion below adds two improvements:

1. I enclosed the entire regex with `()` so that everything captured by this regex is in the same capture group (you can read more about this by searching for "(...)" [in the docs](https://docs.python.org/3/library/re.html))
2. I updated each of the capture groups you had to be `(?...)` instead of `(...)` which means that they are non-capturing groups (you can read more about non-capturing groups by searching for "(?...)" [in the docs](https://docs.python.org/3/library/re.html)). This is because we only want to capture the complete ip address and don't care about the individual components.


```suggestion
pattern = re.compile(r"((?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?),(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?),(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?),(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))")
```